nomad/state/state_store.go

```diff
@@ -76,10 +77,14 @@ func NewStateStore(config *StateStoreConfig) (*StateStore, error) {
 	// Create the state store
 	s := &StateStore{
 		logger:    config.Logger.Named("state_store"),
-		db:        db,
 		config:    config,
 		abandonCh: make(chan struct{}),
 	}
+	s.db = &changeTrackerDB{
+		db:             db,
+		publisher:      event.NewPublisher(),
+		processChanges: processDBChanges,
+	}
```

Comment:
with this much fanciness required to make a `StateStore` it's probably time to use a constructor making sure the necessary things get passed in and set